packages/owl-bot/src/copy-code.ts

```diff
@@ -42,29 +42,67 @@ const readFileAsync = promisify(readFile);
 /**
  * Composes a link to the source commit that triggered the copy.
  */
-function sourceLinkFrom(sourceCommitHash: string): string {
+export function sourceLinkFrom(sourceCommitHash: string): string {
   return `https://github.com/googleapis/googleapis-gen/commit/${sourceCommitHash}`;
 }
 
 /**
- * Copies the code from googleapis-gen to the dest repo, and creates a
- * pull request.
+ * Composes the line we append to a pull request body.
+ */
+export function sourceLinkLineFrom(sourceLink: string): string {
+  return `Source-Link: ${sourceLink}\n`;
+}
+
+/**
+ * Finds the source link in a pull request body.  Returns the empty string if
+ * not found.
+ */
+export function findSourceHash(prBody: string): string {
+  const match =
+    /https:\/\/github.com\/googleapis\/googleapis-gen\/commit\/([0-9A-Fa-f]+)/.exec(
```

Comment:
nit: sometimes I find it's worth using named capture groups for clarity:


```js
/https:\/\/github.com\/googleapis\/googleapis-gen\/commit\/(?<hash>[0-9A-Fa-f]+)/
return match ? match.hash : '';
```

Reply:
Agreed.  This regular expression has a single capture group, so I don't see much potential for confusion.